### PR TITLE
Add lineage db

### PR DESCRIPTION
**Pull request by cdebourcy**

### CODE REVIEW

File: db/migrate/20171004210243_create_taxon_lineages.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class CreateTaxonLineages < ActiveRecord::Migration[5.1]
+  def change
+    create_table :taxon_lineages do |t|
+      t.integer :taxid
```
Comment (ryanking) [inline db/migrate/20171004210243_create_taxon_lineages.rb]: this should probably be not null, yeah?

File: db/migrate/20171004210243_create_taxon_lineages.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      t.integer :superkingdom_taxid
```
Comment (ryanking) [inline db/migrate/20171004210243_create_taxon_lineages.rb]: probably this too?

File: db/migrate/20171004210452_create_taxon_names.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreateTaxonNames < ActiveRecord::Migration[5.1]
+  def change
+    create_table :taxon_names do |t|
+      t.integer :taxid
```
Comment (ryanking) [inline db/migrate/20171004210452_create_taxon_names.rb]: probably should be not null

File: db/migrate/20171004210243_create_taxon_lineages.rb (hunk 1 above)
Comment (cdebourcy) [reply db/migrate/20171004210243_create_taxon_lineages.rb]: agree

File: db/migrate/20171004210243_create_taxon_lineages.rb (hunk 2 above)
Comment (cdebourcy) [reply db/migrate/20171004210243_create_taxon_lineages.rb]: sure

File: db/migrate/20171004210452_create_taxon_names.rb (hunk 3 above)
Comment (cdebourcy) [reply db/migrate/20171004210452_create_taxon_names.rb]: ok

File: ncbitax2lin/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# NCBItax2lin
```
Comment (yunfangjuan) [inline ncbitax2lin/README.md]: should we put all the code here?  could you just put ncbitax2lin into Dockerfile? and maybe write a quick script to run the lineage?

Comment (cdebourcy) [reply ncbitax2lin/README.md]: ok, I forked ncbitax2lin and made it a separate repo that gets cloned inside the rake task

File: db/schema.rb [hunk 5]
```diff
@@ -150,6 +150,26 @@
     t.index ["pipeline_output_id"], name: "index_taxon_counts_on_pipeline_output_id"
   end
 
+  create_table "taxon_lineages", force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci" do |t|
+    t.integer "taxid", null: false
+    t.integer "superkingdom_taxid", null: false
+    t.integer "phylum_taxid"
```
Comment (yunfangjuan) [inline db/schema.rb]: maybe you can make them all not null?

Comment (cdebourcy) [reply db/schema.rb]: yep!

**Pull request merged**